packages/spark/components/masthead.js

```diff
@@ -13,9 +15,54 @@ const addClassOnScroll = (element, scrollPos, scrollPoint, classToToggle) => {
   }
 };
 
+/*
+* Add or remove the class depending
+* on if the user is scrolling up or down
+*/
+const toggleMenu = scrollDirection => {
+  const masthead = document.querySelector('[data-sprk-masthead]');
+  if (scrollDirection === 'down') {
+    masthead.classList.add('sprk-c-Masthead--hidden');
+  } else {
+    masthead.classList.remove('sprk-c-Masthead--hidden');
+  }
+};
+
+/*
+* Set initial scroll direction
+* If the scroll direction changes
+* toggle the masthead visibility 
+*/
+let direction = scrollYDirection();
+const checkScrollDirection = () => {
```

Comment:
Can you debounce this? I think we're already using lodash and so it would be a matter of wrapping the anonymous in this function https://lodash.com/docs/4.17.11#debounce

Debounce for 50 or 100ms whichever feels like it still gives you the responsiveness you need.